builtin/logical/database/path_config_connection.go

```diff
@@ -31,6 +31,8 @@ type DatabaseConfig struct {
 	RootCredentialsRotateStatements []string `json:"root_credentials_rotate_statements" structs:"root_credentials_rotate_statements" mapstructure:"root_credentials_rotate_statements"`
 
 	PasswordPolicy string `json:"password_policy" structs:"password_policy" mapstructure:"password_policy"`
+
+	DisableEscaping bool `json:"disable_escaping" structs:"disable_escaping" mapstructure:"disable_escaping"`
```

Comment:
From what I've observed, it looks like all configuration parameters that are consumed on the plugin side (via [sql.go#L43](https://github.com/hashicorp/vault/blob/main/sdk/database/helper/connutil/sql.go#L43)) don't get added to this list of fields. Examples of those fields are `connection_url`, `username`, `password`, etc. (defined in [sql.go#L23-L28](https://github.com/hashicorp/vault/blob/main/sdk/database/helper/connutil/sql.go#L23-L28)). I'm thinking we could treat `disable_escaping` similarly and remove this?

If we do remove this from `DatabaseConfig`, then we can also remove the `FieldSchema` entry.